src/rdkafka_cgrp.c

```diff
@@ -5173,10 +5202,45 @@ rd_kafka_cgrp_subscription_set(rd_kafka_cgrp_t *rkcg,
                                rd_kafka_topic_partition_list_t *rktparlist) {
         int32_t new_subscription_version =
             rd_atomic32_add(&rkcg->rkcg_subscription_version, 1);
+
+        rkcg->rkcg_flags &= ~(RD_KAFKA_CGRP_F_SUBSCRIPTION |
+                              RD_KAFKA_CGRP_F_WILDCARD_SUBSCRIPTION);
+        if (rkcg->rkcg_subscription)
+                rd_kafka_topic_partition_list_destroy(rkcg->rkcg_subscription);
```

Comment:
Let's use `RD_IF_FREE` here too
```suggestion
        RD_IF_FREE(rkcg->rkcg_subscription,
                   rd_kafka_topic_partition_list_destroy);
```

Reply:
Was already there. Doing the change.